### PR TITLE
Refactor check control

### PR DESCRIPTION
**Pull request by jonman**
- `rb-check-control` now renders a single checkbox input.
- `rb-check-control-group` will render multiple checkboxes.
- `options` attribute replaces `ng-model` to define checkbox content.
- `ng-model` will now return an array of currently checked values.

TP: https://rockabox.tpondemand.com/entity/9534

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased][unreleased]
 
-### Added
+## Changed
```
Comment (ohskylab) [inline CHANGELOG.md]: Should be `###` (I can't believe I'm posting such a pernickety comment and on a Friday evening as well - sorry!)

Comment (jonman) [inline CHANGELOG.md]: Haha, its a fair point! Will update

File: test/unit/rb-check-control/rb-check-control-group.spec.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                isolatedScope = element.isolateScope();
+            };
+        }));
+
+        describe('checkedOptions', function () {
+            beforeEach(function () {
+                compileTemplate('<rb-check-control-group options="options" ng-model="ngModel">' +
+                    '</rb-check-control-group>');
+            });
+
```
Comment (k0nG) [inline test/unit/rb-check-control/rb-check-control-group.spec.js]: Can we check that it renders the right amount of checkboxes here and sets values correctly?

File: src/rb-check-control/rb-check-control-group-directive.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                ngModel: '=',
+                options: '=',
+                title: '@'
+            },
+            restrict: 'E',
+            replace: true,
+            template: template,
+            link: function ($scope, $element, $attributes) {
+
+                $scope.$watch('options|filter:{checked:true}', function (newOptions) {
```
Comment (k0nG) [inline src/rb-check-control/rb-check-control-group-directive.js]: Neat. :+1:

File: src/rb-check-control/rb-check-control-select-all-directive.js [hunk 4]
```diff
@@ -30,12 +30,12 @@ define([
 
         return {
             scope: {
+                enableSelectAll: '@',
```
Comment (k0nG) [inline src/rb-check-control/rb-check-control-select-all-directive.js]: Why does the select all need this?

Comment (jonman) [inline src/rb-check-control/rb-check-control-select-all-directive.js]: Currently the select all directive also renders a title without the checkbox. I could move that out into the group directive instead and use an `ng-if` to toggle which will fix this (bug)[https://rockabox.tpondemand.com/entity/9495]

File: test/unit/rb-check-control/rb-check-control-group.spec.js (hunk 2 above)
Comment (jonman) [inline test/unit/rb-check-control/rb-check-control-group.spec.js]: Will update.

File: src/rb-check-control/rb-check-control-select-all-directive.js (hunk 4 above)
Comment (k0nG) [inline src/rb-check-control/rb-check-control-select-all-directive.js]: Yeah move that into a `label` tag inside the group directive.

**Comment by k0nG** [comment]
MOG

**Pull request merged**